buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -1130,6 +1149,11 @@ public int setBytes(int index, InputStream in, int length) throws IOException {
             ByteBuf s = c.buf;
             int adjustment = c.offset;
             int localLength = Math.min(length, s.capacity() - (index - adjustment));
+            if (localLength == 0) {
```

Comment:
The codes are still correct without these lines. However, it can avoid calling `setBytes` for empty buffers.